test/metabase/search/fulltext/scoring_test.clj

```diff
@@ -125,9 +138,101 @@
+      [{:model "card" :id 1 :name "card well known" :view_count 10}
+       {:model "card" :id 2 :name "card famous"     :view_count 100}
+       {:model "card" :id 3 :name "card popular"    :view_count 50}]
+      (is (= [["card" 2 "card famous"]
+              ["card" 3 "card popular"]
+              ["card" 1 "card well known"]]
+             (search :view-count "card")))))
+
+  (testing "don't error on fresh instances with no view count"
+    (with-index-contents
+      [{:model "card"      :id 1 :name "view card"      :view_count 0}
+       {:model "dashboard" :id 2 :name "view dashboard" :view_count 0}
+       {:model "dataset"   :id 3 :name "view dataset"   :view_count 0}]
+      (is (= [["dashboard" 2 "view dashboard"]
+              ["card"      1 "view card"]
+              ["dataset"   3 "view dataset"]]
+             (search* :view-count "view"))))))
+
+(deftest view-count-edge-case-test
+  (testing "view count max out at p99, outlier is not preferred"
+    (when (search/supports-index?)
+      (let [table-name (search.index/random-table-name)]
+        (binding [search.index/*active-table* table-name]
+          (search.index/create-table! table-name)
+          (mt/with-model-cleanup [:model/Card]
+            (let [search-term    "view-count-edge-case"
+                  card-with-view #(merge (mt/with-temp-defaults :model/Card)
+                                         {:name search-term
+                                          :view_count %})
+                  _               (t2/insert! :model/Card (concat (repeat 20 (card-with-view 0))
+                                                                  (for [i (range 1 80)]
+                                                                    (card-with-view i))))
+                  outlier-card-id (t2/insert-returning-pk! :model/Card (card-with-view 100000))
+                  _               (#'search.ingestion/batch-update!
+                                   (#'search.ingestion/spec-index-reducible "card" [:= :this.name search-term]))
+                  first-result-id (-> (search* :view-count search-term) first)]
+              (is (some? first-result-id))
+              (is (not= outlier-card-id first-result-id)))))))))
+
+(deftest ^:parallel dashboard-count-test
+  (testing "cards used in dashboard have higher rank"
+    (with-index-contents
+      [{:model "card" :id 1 :name "card no used" :dashboardcard_count 2}
+       {:model "card" :id 2 :name "card used" :dashboardcard_count 3}]
+      (is (= [["card" 2 "card used"]
+              ["card" 1 "card no used"]]
+             (search :dashboard "card")))))
+
+  (testing "it has a ceiling, more than the ceiling is considered to be equal"
+    (with-index-contents
+      [{:model "card" :id 1 :name "card" :dashboardcard_count 22}
+       {:model "card" :id 2 :name "card" :dashboardcard_count 11}]
+      (is (= [["card" 1 "card"]
+              ["card" 2 "card"]]
+             (search* :dashboard "card"))))))
```

Comment:
Same here about potential flaking in the future.

I'm thinking perhaps search should break ties based on name, in the db query 🤔